### ISSUE TITLE
Docker plain quickstarter is unable to create "cd-user-bitbucket-token" secret

### ISSUE THREAD

**Issue by felipecruz91** [opendevstack/ods-quickstarters#477]
**Describe the bug**
When creating a new Dockerplain quickstarter with the Provisioning App, the Jenkins pipeline fails because it cannot create the `cd-user-bitbucket-token` secret in the `-cd` namespace.

**To Reproduce**
1. Create a Docker plain quickstarter through the Provisioning App.
2. Go to Jenkins and check the pipeline that builds the `master` branch.
3. Go to the Jenkins build logs.
4. See error (log output further below).

**Expected behavior**
The Jenkins pipeline should run successfully and create the `cd-user-bitbucket-token` secret in their -cd namespace if it not exists.

**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack 3.x

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
[Pipeline] {
[Pipeline] withCredentials
[Pipeline] // withCredentials
[Pipeline] echo
Secret cd-user-bitbucket-token does not exist yet, it will be created now.
[Pipeline] withCredentials
Masking supported pattern matches of $USERNAME or $PASSWORD
[Pipeline] {
[Pipeline] echo
Requesting token via PUT https://bitbucket.biscrum.com/rest/access-tokens/1.0/users/**** with payload={"name": "ods-jenkins-shared-library-****-cd", "permissions": ["PROJECT_WRITE", "REPO_WRITE"]}
[Pipeline] sh
+ set +x
[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] stage
[Pipeline] { (odsPipeline error)
[Pipeline] echo
WARN: [test] ***** Finished ODS Pipeline for test (with error) ***** (took 44008 ms)
```

**Additional context**
Issue comes from #471

**Comment by jssnull** [comment]
https://github.com/opendevstack/ods-quickstarters/issues/472
This issue could be related

**Comment by michaelsauter** [comment]
@felipecruz91 is this reproducible? If you run the pipeline again, does the problem occur again? Is this only with this specific project or also with others?

Right now it does not look related to #472 to me.

**Comment by felipecruz91** [comment]
@michaelsauter Yes, it is reproducible (it will keep failing with the same error over and over again). For the Dockerplain quickstarter or any other to work successfully, the project in OpenShift must contain the `cd-user-bitbucket-token` secret in their -cd namespace. If I create it manually in OpenShift, then the Jenkins pipeline builds successfully.

**Comment by michaelsauter** [comment]
Very odd. Same in another namespace?

It would be awesome if you can run again with more debug output. Sadly, the `curl` command is missing this in current `master`.  The open PR #462 is adding error output, see https://github.com/opendevstack/ods-jenkins-shared-library/pull/462/files#diff-9d140594f5d584cb785653f24b84bc4e0e9976f0c28745f6815a6d30d4ba6fdcR359. With this, the problem should be printed.

**Comment by felipecruz91** [comment]
@michaelsauter Can you double check the PR number in the previous comment? I cannot find the info you mentioned in that PR.

**Comment by michaelsauter** [comment]
@felipecruz91 The scrolling is messed up a little. Check the file `BitbucketService.groovy`. It contains a change from `curl --silent` to `curl -sS`.

**Comment by clemensutschig** [comment]
@felipecruz91  - can we please get the log (current +5 lines) - I have seen an issue like this, but I remember that to be fixed ...

**Comment by michaelsauter** [comment]
@felipecruz91 #462 has been merged in the meantime, so if you run your pipeline again with latest master you should hopefully see the reason for the error.

**Comment by michaelsauter** [comment]
@felipecruz91 could this have been a temporary issue with the Bitbucket instance? Is this still happening? Should we close this?

**Comment by felipecruz91** [comment]
@michaelsauter I haven't managed either to replicate the issue or seeing this happening again since it was originally reported. Let's close it and will open it again if this comes back.

**Comment by SHoen** [comment]
I stumbled over the same error and noticed that the generated Basic Auth Header of the token request was incorrect.

I updated the password of the bitbucket user in the yml of `cd-user-with-password` with Base64 encoded string generated by bash: `$ echo myfancypass | base64 ` Unfortunately it added a newline in the encoded string which lead to this error. 

Using `echo -n myfancypass | base64 ` instead did the trick for me.
